perfkitbenchmarker/beam_pipeline_options.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2017 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import yaml
+
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import kubernetes_helper
+
+FLAGS = flags.FLAGS
+
+
+def GetStaticPipelineOptions(options_list):
+  options = []
+  for option in options_list:
+    if not len(option.keys()) == 1:
+      raise Exception('static_pipeline_options should only have 1 key/value')
+    option_kv = option.items()[0]
+    options.append((option_kv[0], option_kv[1]))
+
+  return options
+
+
+def EvaluateDynamicPipelineOptions(dynamic_options):
+  """
+  Takes the user's dynamic args and retrieves the information to fill them in.
+
+  dynamic_args is a python map of argument name -> {type, kubernetesSelector}
+  returns a list of tuples containing (argName, argValue)
+  """
+  filledOptions = []
+  for optionDescriptor in dynamic_options:
+    fillType = optionDescriptor['type']
+    optionName = optionDescriptor['name']
+
+    if not fillType:
+      raise errors.Config.InvalidValue(
+          'For dynamic arguments, you must provide a "type"')
+
+    if fillType == 'NodePortIp':
+      argValue = RetrieveNodePortIp(optionDescriptor)
+    elif fillType == 'LoadBalancerIp':
+      argValue = RetrieveLoadBalancerIp(optionDescriptor)
+    elif fillType == 'TestValue':
+      argValue = optionDescriptor['value']
+    else:
+      raise errors.Config.InvalidValue(
+          'Unknown dynamic argument type: %s' % (fillType))
+
+    filledOptions.append((optionName, argValue))
+
+  return filledOptions
+
+
+def GenerateAllPipelineOptions(it_args, it_options, static_pipeline_options,
+                               dynamic_pipeline_options):
+  """
+  :param it_args: options list passed in via FLAGS.beam_it_args
+  :param it_options: options list passed in via FLAGS.beam_it_options
+  :param static_pipeline_options: options list passed in via
+      benchmark_spec.dpb_service.static_pipeline_options
+  :param dynamic_pipeline_options: options list passed in via
+      benchmark_spec.dpb_service.dynamic_pipeline_options
+  :return: a list of values of the form "\"--option_name=value\""
+  """
+  # beam_it_options are in [--option=value,--option2=val2] form
+  user_option_list = []
+  if it_options is not None and len(it_options) > 0:
+    user_option_list = it_options.rstrip(']').lstrip('[').split(',')
```

Comment:
I think long term we can move this to a utility, seems like a useful parsing snippet. Thanks